### ISSUE TITLE
Getting duplicate random data unless using unique

### ISSUE THREAD

**Issue by spawnia** [fzaninotto/Faker#1417]
### Setup
I use Faker for seeding my database in Laravel. I define my factory method as follows (simplified):

````php
$factory->define(App\Textfield::class, function (Faker $faker) {
    $faker->seed(1234);
    
    return [
        'name' => $faker->company,
        'x' => $faker->randomDigit,
        'effects' => $faker->randomElement(['b', 's', 'i', 'n', 'u', 'l', 'z', 'k', 'v', 'qn', 'hn', 'mn']),
    ];
});
````

I call this 10 times like this `factory(App\Textfield::class, 10)->create();`

### Problem
Faker just spits out 10 rows of the same data, among all columns. When i change one of the calls to be unique, such as `$faker->unique()->company`, the company name gets generated uniquely, as well as all the other values.

### What i would like to happen
The data gets created randomly, without me having to specify unique on one of the columns. I do not need the columns to be unique, i just want the values to be somewhat randomly distributed. Maybe this is a problem with Laravel though - i thought about calling the factory method in a loop, incrementing the seed with each iteration. I do not see an option of passing this to the factory method though.

**Comment by tomzx** [comment]
The issue is likely to be that you always use the same seed. Thus, every call to your closure will result in the same data being generated.

Why do you need to specify the seed?

**Comment by spawnia** [comment]
I would like to always get the same data when populating my database.

What happens here is expected behaviour, i would not even call it a bug. But somehow, using `unique()` on just one of the columns solves my problem. I was wondering if there is to generally apply the technique that unique uses to all of my generated data - ensuring to always get the same result on each run but with different tuples, allowing duplicate values.

**Comment by fzaninotto** [comment]
Can you rephrase? I fail to understand your need.

**Comment by PatrickRose** [comment]
I think what you want is to always get the same values when you run the seeder

In which case, set the seed _before_ you call the method 10 times instead of during the method.

**Comment by spawnia** [comment]
@fzaninotto A more concrete example of what i am experiencing:

### Current behaviour
````
$factory->define(App\Textfield::class, function (Faker $faker) {
    $faker->seed(1234);
    
    return [
        'name' => $faker->company,
        'number' => $faker->randomDigit
    ];
});
... (in another file)
factory(App\Textfield::class, 3)->create();
````
Example output:

| name | number |
| ------ | --------- |
| Foo    | 2 |
| Foo    | 2 |
| Foo    | 2 |

Using unique on one of the columns:
````
$factory->define(App\Textfield::class, function (Faker $faker) {
    $faker->seed(1234);
    
    return [
        'name' => $faker->unique()->company,
        'number' => $faker->randomDigit
    ];
});
... (in another file)
factory(App\Textfield::class, 3)->create();
````
Example output:

| name | number |
| ------ | --------- |
| Foo    | 2 |
| Bar    | 6 |
| Baz    | 8 |

So, using unique here not only causes the first column to generate uniquely but also the second to at least get different values.

### Wanted behaviour
````
$factory->define(App\Textfield::class, function (Faker $faker) {
    $faker->seed(1234);
    
    return [
        'name' => $faker->company,
        'number' => $faker->randomDigit
    ];
});
... (in another file)
factory(App\Textfield::class, 3)->create();
````
Example output:

| name | number |
| ------ | --------- |
| Foo    | 2 |
| Bar    | 4 |
| Foo    | 8 |

I would like to be able to generate those values in a way that:
- always yields the same result (using a seed)
- does not spit out rows that are just the same
- allow duplicate values to occur

What @PatrickRose suggest should fix the issue i am having, although it would require some rework in the code.

I brought this up because of the particular case of what calling `->unique()` on one of the columns does, maybe there is a way to use that behaviour of semi-randomness but allow duplicates between multiple calls. I also see that what i am suggesting is a bit of a corner case, so i also would not mind the issue being closed - but if someone wants to take a stab at it that would be nice, too,

**Comment by PatrickRose** [comment]
I'm not too familiar with Faker internals but I believe what's happening is
calling unique() moves the random number generator along until it gets a
unique value which is why you're seeing new values on each iteration.

To be honest, what you're trying to do seems a bit off - why do you want
faker to return the same values on each run? The idea is to get random data
so you can spot bugs.

**Comment by spawnia** [comment]
There are many valid reasons why you would always want to generate the same data. Especially for automated testing - it can be beneficial to work with a set of fixed inputs so you can test the outputs.

In my particular case, i am designing an API and am doing some transformations on the generated data before exposing it. And that i can only test when dealing with known base information.

**Comment by spawnia** [comment]
Closing this issue since it is possible to get the wanted behaviour, it just requires a bit of a workaround from the laravel code.

Faker works as intended.
